Geometry/CaloGeometry/interface/CaloGeometry.h

```diff
@@ -31,10 +32,10 @@ class CaloGeometry
 			      const CaloSubdetectorGeometry* geom    ) ;
     
       /// Get the position of a given detector id
-      const GlobalPoint& getPosition( const DetId& id ) const ;
+      const GlobalPoint& getPosition( const DetId& id );
```

Comment:
This needs to be changed back to `getPosition( const DetId& id ) const`.

It is the remove of `const` from these member functions which caused you to start to do so many casts to remove `const`.